src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/generator/AbstractQueryGenerator.java

```diff
@@ -70,8 +76,52 @@ private String generateQueryBody(@NonNull Criteria criteria, @NonNull List<Pair<
     @NonNull
     protected Pair<String, List<Pair<String, Object>>> generateQueryBody(@NonNull DocumentQuery query) {
         final List<Pair<String, Object>> parameters = new ArrayList<>();
-        final String queryString = this.generateQueryBody(query.getCriteria(), parameters);
+        String queryString = this.generateQueryBody(query.getCriteria(), parameters);
+
+        if (StringUtils.hasText(queryString)) {
+            queryString = "WHERE" + " " + queryString;
```

Comment:
Do we need prepend whitespace before `WHERE` to avoid caller missing whitespace in the end?